### PR TITLE
planner: not push tiflash if extra phys tbl id is needed

### LINKED ISSUE #32829: TiFlash throw `DB::String TiDB::TableInfo::getColumnName(const DB::ColumnID) const: Invalidate column id -3 for table t_137` error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
mysql> show create table employees;
+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table     | Create Table                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| employees | CREATE TABLE `employees` (
  `id` int(11) NOT NULL,
  `fname` varchar(30) DEFAULT NULL,
  `lname` varchar(30) DEFAULT NULL,
  `hired` date NOT NULL DEFAULT '1970-01-01',
  `separated` date DEFAULT '9999-12-31',
  `job_code` int(11) DEFAULT NULL,
  `store_id` int(11) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY RANGE (`store_id`)
(PARTITION `p0` VALUES LESS THAN (6),
 PARTITION `p1` VALUES LESS THAN (11),
 PARTITION `p2` VALUES LESS THAN (16),
 PARTITION `p3` VALUES LESS THAN (21)) |
+-----------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> insert into employees values(100,'aa','aa','2020-01-01',null,10,10);
Query OK, 1 row affected (0.00 sec)

mysql> select count(*) from employees;
ERROR 1105 (HY000): DB::String TiDB::TableInfo::getColumnName(const DB::ColumnID) const: Invalidate column id -3 for table t_137
```
### 2. What did you expect to see? (Required)
query finished without error
### 3. What did you see instead (Required)
ERROR 1105 (HY000): DB::String TiDB::TableInfo::getColumnName(const DB::ColumnID) const: Invalidate column id -3 for table t_137
### 4. What is your TiDB version? (Required)
tidb @ 74d74b0a7b6bee275d392199bb319859682627ff
<!-- Paste the output of SELECT tidb_version() -->

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #32829

Problem Summary:

### What is changed and how it works?
The root cause is TiFlash does not support virtual column `ExtraPhysTblID` yet.
This pr explicitly remove TiFlash's access path if `ExtraPhysTblID` is needed.
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/tiflash_test.go [hunk 1]
```diff
@@ -1071,3 +1071,54 @@ func (s *tiflashTestSuite) TestForbidTiflashDuringStaleRead(c *C) {
...
+	rows := tk.MustQuery("explain select count(*) from t").Rows()
+	resBuff := bytes.NewBufferString("")
+	for _, row := range rows {
+		fmt.Fprintf(resBuff, "%s\n", row)
+	}
+	res := resBuff.String()
+	c.Assert(strings.Contains(res, "tiflash"), IsTrue)
+	c.Assert(strings.Contains(res, "tikv"), IsFalse)
+
+	rows = tk.MustQuery("explain select count(*) from t for update").Rows()
```
Comment (mjonss) [inline executor/tiflash_test.go:1096]: Do you know where in the planner this is decided to never go to tiflash?

**Review by mjonss — APPROVED** [review]
LGTM

Just a question about SelectLock/SELECT FOR UPDATE, maybe we should at least add a comment where the plan is made why it would not use TiFlash? (OK to ignore)

File: executor/tiflash_test.go (hunk 1 above)
Comment (windtalker) [reply executor/tiflash_test.go:1096]: It's [here](https://github.com/pingcap/tidb/blob/74d74b0a7b6bee275d392199bb319859682627ff/planner/optimize.go#L114).

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**